gzhttp/compress.go

```diff
@@ -464,6 +464,12 @@ func NewWrapper(opts ...option) (func(http.Handler) http.HandlerFunc, error) {
 	return func(h http.Handler) http.HandlerFunc {
 		return func(w http.ResponseWriter, r *http.Request) {
 			w.Header().Add(vary, acceptEncoding)
+			if c.allowCompressedRequests && contentGzip(r) {
+				r.Header.Del(contentEncoding)
+				r.Body = io.NopCloser(&gzipReader{body: r.Body})
```

Comment:
No need to wrap. `*gzipReader` is an `io.ReadCloser` that will close the body. Also it doesn't leak the decoder.

```suggestion
				r.Body = &gzipReader{body: r.Body}
```